target_ws_pkgs.repos

```diff
@@ -19,108 +19,119 @@ repositories:
     type: git
     url: https://github.com/ros2/rmw_implementation.git
-    version: 2.8.2
+    version: 2.15.3
+  # needed because of https://github.com/ros2/rcl_interfaces/issues/144
   ros2/rosidl:
     type: git
     url: https://github.com/ros2/rosidl.git
-    version: 3.1.5
+    version: 4.6.4
+  ros2/rosidl_core:
+    type: git
+    url: https://github.com/ros2/rosidl_core.git
+    version: 0.2.0
   ros2/rosidl_dds:
     type: git
     url: https://github.com/ros2/rosidl_dds.git
-    version: 0.8.1
+    version: 0.11.1
   ros2/rosidl_defaults:
     type: git
     url: https://github.com/ros2/rosidl_defaults.git
-    version: 1.2.0
+    version: 1.6.0
+  ros2/rosidl_dynamic_typesupport:
+    type: git
+    url: https://github.com/ros2/rosidl_dynamic_typesupport.git
+    version: 0.1.2
 
   uros/rcl:
     type: git
     # this is a regular fork, but renamed due to potential clashes with ros2/rcl
     url: https://github.com/yaskawa-global/micro_ros_rcl.git
-    # NOTE: the fork adds additional patches, on top of 'upstream/humble'.
-    #       Branch name: 'humble-motoplus1-candidate-20230624'
-    version: 1423adabdf10cd3165244e77479763d44bf5945a
+    # NOTE: the fork adds 6 additional patches, on top of 'upstream/jazzy'.
+    # Branch name: 'jazzy'
+    version: 032fee21bcbd75ad0a8b43fe5f1a6c983d5f0d34
   uros/rcutils:
     type: git
     url: https://github.com/yaskawa-global/rcutils.git
-    # 'upstream/humble' (couple commits past 5.1.3), with 5 patches
-    version: c8ec6220ae77d3363de4c377485fd9c653be2936
+    # 'upstream/jazzy' (couple commits past 6.7.2), with 6 patches
+    #  Branch name: 'jazzy'
+    version: 0512f00dc515c1866eec5dc732e10f41228b707f
   uros/micro_ros_utilities:
     type: git
     url: https://github.com/micro-ROS/micro_ros_utilities.git
-    version: 3.0.1
+    version: 5.0.1
   uros/rmw-microxrcedds:
     type: git
     url: https://github.com/micro-ROS/rmw-microxrcedds.git
-    version: 3.0.2
+    version: 5.0.1
   uros/rosidl_typesupport:
     type: git
     url: https://github.com/yaskawa-global/rosidl_typesupport.git
-    # 'upstream/humble' with a single patch of our own
-    version: fea3ab8a497b87b04d489d7cfb733a8455b90fad
+    # 'upstream/jazzy' with a single patch of our own
+    #  Branch name: 'jazzy'
+    version: 79885fc1a3e73e0e4205475e7e3804bba73e1e1d
   uros/rosidl_typesupport_microxrcedds:
     type: git
     url: https://github.com/micro-ROS/rosidl_typesupport_microxrcedds.git
-    version: 3.0.1
+    version: 5.0.1
+
+
 
   uros/tracetools:
     type: git
     url: https://github.com/ros2/ros2_tracing.git
-    version: 4.1.1
+    version: 8d898b5ff0d79ef34eeae73a6ef3a20b30fe1ac8
+    # NOTE: this is not the same as the jazzy release. 
+    # this includes a few commits past jazzy. This specific
+    # commit contains a necessary change (getting rid of the
+    # deprecation message macros). We could cherry pick this
+    # commit instead, but then we would need a new fork. 
```

Comment:
@jimmy-mcelwain: looking at the commit history, the commit referenced here (https://github.com/ros2/ros2_tracing/commit/8d898b5ff0d79ef34eeae73a6ef3a20b30fe1ac8) only appears on the `rolling` branch. I don't see it on [jazzy](https://github.com/ros2/ros2_tracing/commits/jazzy).

Github also lists it as appearing in `rolling` tags/releases:

![image](https://github.com/user-attachments/assets/5ab6f6fe-1afc-4c76-99cf-670e1ec0b9f8)

?

Reply:
Yeah, my comment there is inaccurate, it's not just a few commits past release, it's on a different branch. If I remember correctly, the deprecation message macros broke the build, and I saw that a commit on rolling got rid of the macros, so I referenced it directly. I was just trying to avoid another fork.

Reply:
After looking into it more, I found that the issue is that the M+ compiler does not like [this line](https://github.com/ros2/ros2_tracing/blob/42e26cca398246db56881e17394abf9f8d3b7755/tracetools/include/tracetools/tracetools.h#L130). 

The `__attribute__((deprecated(msg)))` form is only supported in GCC version 4.5+, which is newer than what we are using.

Reply:
I could create a new fork for this, if you think it is warranted @gavanderhoorn. The change would be very straightforward. And since the `__attribute__((deprecated(msg)))` form is removed from rolling, we would likely only need the fork for Jazzy.

Reply:
I'm in favor of Jimmy's approach of avoiding another fork. 

Since the update doesn't apply any functionality change to the code, targeting the 'wrong' branch is ok, as long we explain clearly why it was done and verify everything still works.